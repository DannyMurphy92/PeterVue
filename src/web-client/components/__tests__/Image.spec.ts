import { shallowMount, Wrapper, createLocalVue } from '@vue/test-utils'
import ImageViewer from '@/components/Image.vue'

const localVue = createLocalVue()

describe('Image', () => {
  describe('GIVEN minimum required props', () => {
    let wrapper: Wrapper<any>

    beforeEach(() => {
      wrapper = shallowMount(ImageViewer, {
        localVue,
        propsData: {
          url: 'https://www.test.com/image.png',
          altText: 'Test alt',
        },
      })
    })

    it('SHOULD render img tag correctly', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
