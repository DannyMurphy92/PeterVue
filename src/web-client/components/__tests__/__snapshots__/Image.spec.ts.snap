// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image GIVEN minimum required props SHOULD render img tag correctly 1`] = `
<div
  class="image-container"
>
  <img
    alt="Test alt"
    class="image-viewer"
    src="https://www.test.com/image.png"
  />
</div>
`;
